exporters/committime/collector_image.py

```diff
@@ -0,0 +1,214 @@
+                metric.commit_time, self._timedate_format
+            )
+        return metric
+
+    def get_commit_time(self, metric) -> Optional[CommitMetric]:
+        return super().get_commit_time(metric)
+
+    def _set_commit_time_from_annotations(
+        self, metric: CommitMetric, errors: list
+    ) -> CommitMetric:
+        if not metric.commit_time:
+            commit_time_annotation = get_env_var(
+                COMMIT_DATE_ANNOTATION_ENV,
+                CommitMetric._ANNOTATION_MAPPIG.get("commit_time"),
+            )
+            commit_time = metric.annotations.get(commit_time_annotation)
+            if commit_time:
+                metric.commit_time = commit_time
+                logging.debug(
+                    "Commit time for image %s provided by '%s' annotation: %s",
+                    metric.image_hash,
+                    commit_time_annotation,
+                    metric.commit_time,
+                )
+            else:
+                errors.append(
+                    "Couldn't get commit time from annotations nor image label"
+                )
+        return metric
+
+    # overrides collector_base.generate_metric()
+    def generate_metrics(self) -> Iterable[CommitMetric]:
+
+        # Initialize metrics list
+        metrics = []
+        app_label = pelorus.get_app_label()
+
+        logging.debug("Searching for images with label: %s" % app_label)
+
+        v1_images = self._kube_client.resources.get(
+            api_version="image.openshift.io/v1", kind="Image"
+        )
+
+        images = v1_images.get(label_selector=app_label)
+
+        images_by_app = self._get_openshift_obj_by_app(images, app_label)
+
+        if images_by_app:
+            metrics += self._get_metrics_by_apps_from_images(images_by_app)
+
+        return metrics
+
+    def _get_metrics_by_apps_from_images(self, images_by_app):
+        metrics = []
+        for app in images_by_app:
+            images = images_by_app[app]
+            for image in images:
+                metric = None
+                errors = []
+
+                try:
+                    metric = self.commit_metric_from_image(app, image, errors)
+                except Exception:
+                    logging.error(
+                        "Cannot collect metrics from image: %s" % (image.metadata.name)
+                    )
+                    raise
+
+                if metric:
+                    if errors:
+                        msg = (
+                            f"Missing data for CommitTime metric from Image "
+                            f"{metric.image_hash} in app {app}: "
+                            f"{'.'.join(str(e) for e in errors)}"
+                        )
+                        logging.warning(msg)
+                        continue
+
+                    logging.debug("Adding metric for app %s" % app)
+                    metrics.append(metric)
```

Comment:
Bit of confusion over this section:

If `commit_metric_from_image` always returns a `CommitMetric`, this will never be falsy. I think this is a holdover from the other exporters, which could return `None` if the commit time couldn't be found.

Reply:
So what do you propose? We only collect metrics which are OK'ish ,same as other exporters.

Reply:
I'm saying the `if metric` is unnecessary, because it's always true.